base/applications/osk/main.c

```diff
@@ -451,34 +452,60 @@ LRESULT APIENTRY OSK_ThemeHandler(HWND hDlg, NMCUSTOMDRAW *pNmDraw)
     */
     if (hTheme)
     {
-        /*
-            The button could be either in normal state or pushed.
-            Retrieve its state and save to a variable.
-        */
-        if (pNmDraw->uItemState & CDIS_DEFAULT)
-        {
-            iState = PBS_DEFAULTED;
-        }
-        else if (pNmDraw->uItemState & CDIS_SELECTED)
+        /* Obtain CDDS drawing stages */
+        switch (pNmDraw->dwDrawStage)
         {
-            iState = PBS_PRESSED;
-        }
-        else if (pNmDraw->uItemState & CDIS_HOT)
-        {
-            iState = PBS_HOT;
-        }
+            case CDDS_PREPAINT:
+            {
+                /*
+                    The button could be either in normal state or pushed.
+                    Retrieve its state and save to a variable.
+                */
+                if (pNmDraw->uItemState & CDIS_DEFAULT)
+                {
+                    iState = PBS_DEFAULTED;
+                }
+                else if (pNmDraw->uItemState & CDIS_SELECTED)
+                {
+                    iState = PBS_PRESSED;
+                }
+                else if (pNmDraw->uItemState & CDIS_HOT)
+                {
+                    iState = PBS_HOT;
+                }
 
-        if (IsThemeBackgroundPartiallyTransparent(hTheme, BP_PUSHBUTTON, iState))
-        {
-            /* Draw the application if the theme is transparent */
-            DrawThemeParentBackground(hDlg, pNmDraw->hdc, &pNmDraw->rc);
-        }
+                if (IsThemeBackgroundPartiallyTransparent(hTheme, BP_PUSHBUTTON, iState))
+                {
+                    /* Draw the application if the theme is transparent */
+                    DrawThemeParentBackground(hDlgButtonCtrl, pNmDraw->hdc, &pNmDraw->rc);
+                }
 
-        /* Draw it */
-        DrawThemeBackground(hTheme, pNmDraw->hdc, BP_PUSHBUTTON, iState, &pNmDraw->rc, NULL);
+                /* Draw it */
+                DrawThemeBackground(hTheme, pNmDraw->hdc, BP_PUSHBUTTON, iState, &pNmDraw->rc, NULL);
+                
+                Ret = CDRF_SKIPDEFAULT;
+                break;
+            }
+
+            case CDDS_PREERASE:
+            {
+                Ret = CDRF_DODEFAULT;
+                break;
+            }
+
+            default:
+                Ret = CDRF_SKIPDEFAULT;
+                break;
+        }
+    }
+    else
+    {
+        /* hTheme is NULL so bail right away */
+        Ret = CDRF_SKIPDEFAULT;
```

Comment:
Hm so if there is no theme how is it actually going to look?

Reply:
Without any theme On-Screen Keyboard will just look normal.

Reply:
Actually it would make more sense to let it do everything the default way when hTheme is null, I don't know how it does work like this